### PR TITLE
Fixed issue #17903: Incorrect IP address used when having multiple forward from proxies

### PR DESCRIPTION
**Pull request by tassoman**
Fixed issue Mantis #17903 - branch 5.x (master)

### CODE REVIEW

File: application/helpers/common_helper.php [hunk 1]
```diff
@@ -4625,25 +4625,38 @@ function ellipsize($sString, $iMaxLength, $fPosition = 1, $sEllipsis = '&hellip;
...
+        //Check IP Address passed from proxy
+        $vComma = strpos($_SERVER['HTTP_X_FORWARDED_FOR'], ',');
+        if (false === $vComma && filter_var($_SERVER['HTTP_X_FORWARDED_FOR'], FILTER_VALIDATE_IP) !== false) {
+            // Single forward 
+            $sIPAddress = $_SERVER['HTTP_X_FORWARDED_FOR'];
+        } else {
+            // Multitple forward
+            $aForwarded = explode(',', $_SERVER['HTTP_X_FORWARDED_FOR']);
+            if (false !== filter_var($aForwarded[0], FILTER_VALIDATE_IP)) {
+                $sIPAddress = $aForwarded[0];
```
Comment (Shnoulle) [inline application/helpers/common_helper.php:4649]: Then it take the first one ?
`127.0.0.1,127.0.0.2,127.0.0.3` return `127.0.0.1`

Seems OK with stackoverflow , but a better source can be great
Adding a source for explanation (rfc ; https://datatracker.ietf.org/doc/html/rfc7239#section-5.2 ?)

**Review by Shnoulle — APPROVED** [review]
Maybe adding a source for comment ?

File: application/helpers/common_helper.php (hunk 1 above)
Comment (tassoman) [reply application/helpers/common_helper.php:4649]: I based my contribution on [Mozilla Http Headers documentation](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-For). 
The validation goes beyond function behaviour.  I guess is done inside `filter_var($var, FILTER_VALIDATE_IP)`

Comment (Shnoulle) [reply application/helpers/common_helper.php:4649]: Yes, not about `FILTER_VALIDATE_IP` just adding the mozilla link on comment about explode ;)

File: application/helpers/common_helper.php [hunk 2]
```diff
@@ -4625,25 +4625,38 @@ function ellipsize($sString, $iMaxLength, $fPosition = 1, $sEllipsis = '&hellip;
...
-        //Check IP address passed from proxy
-        $sIPAddress = $_SERVER['HTTP_X_FORWARDED_FOR'];
+    } elseif (!empty($_SERVER['HTTP_X_FORWARDED_FOR'])) {
+        //Check IP Address passed from proxy
+        $vComma = strpos($_SERVER['HTTP_X_FORWARDED_FOR'], ',');
+        if (false === $vComma && filter_var($_SERVER['HTTP_X_FORWARDED_FOR'], FILTER_VALIDATE_IP) !== false) {
+            // Single forward 
+            $sIPAddress = $_SERVER['HTTP_X_FORWARDED_FOR'];
+        } else {
+            // Multitple forward
```
Comment (Shnoulle) [inline application/helpers/common_helper.php:4646]: `// Multitple forward , see https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-For`

for example ?

Comment (tassoman) [reply application/helpers/common_helper.php:4646]: can I push another commit to this branch-pr ? :thinking:

Comment (Shnoulle) [reply application/helpers/common_helper.php:4646]: Why not ? I can do if you want ;)

Comment (Shnoulle) [reply application/helpers/common_helper.php:4646]: You can locally, or if you want : you can edit this file online for examle

Comment (tassoman) [reply application/helpers/common_helper.php]: I've pushed both! Thank you

**Review by Shnoulle — APPROVED** [review]

**Comment by tassoman** [comment]
@Shnoulle , maybe the `strpos()` part can be archived returning always 1st `explode()` element.
if the separator is not present, [the entire string is returned as array element](https://www.php.net/manual/en/function.explode.php). 🤔

**Pull request merged**